File: app/src/main/java/com/innso/mobile/ui/activities/LoginActivity.java
package com.innso.mobile.ui.activities;

import androidx.databinding.DataBindingUtil;
import android.os.Bundle;

import com.innso.mobile.R;
import com.innso.mobile.databinding.ActivityLoginBinding;
import com.innso.mobile.ui.viewModels.LoginViewModel;

public class LoginActivity extends BaseActivity {

    private ActivityLoginBinding binding;

    private LoginViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_login);
        viewModel = new LoginViewModel();
        binding.setViewModel(viewModel);
    }

    @Override
    protected void onResume() {
        super.onResume();
        subscribe();
    }

    public void subscribe(){
        disposable.add(viewModel.observableSnackBar().subscribe(event -> showError(binding.getRoot(), event.getMessage())));
        disposable.add(viewModel.observableStartActivity().subscribe(this::startActivity));
    }
}

File: app/src/main/java/com/innso/mobile/ui/activities/LoginActivity.kt
package com.innso.mobile.ui.activities

import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProviders
import com.innso.mobile.R
import com.innso.mobile.databinding.ActivityLoginBinding
import com.innso.mobile.viewModels.onBoarding.LoginViewModel

class LoginActivity : BaseActivity() {

    private lateinit var binding: ActivityLoginBinding

    private lateinit var viewModel: LoginViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        initViewModel()
    }

    private fun initViewModel() {
        viewModel = ViewModelProviders.of(this).get(LoginViewModel::class.java)
        binding.viewModel = viewModel
        binding.lifecycleOwner = this
        subscribeViewModel(viewModel, binding.root)
    }
}
